src/develop/develop.h

```diff
@@ -317,26 +332,10 @@ typedef struct dt_develop_t
     GtkWidget *floating_window, *softproof_button, *gamut_button;
   } profile;
 
-  // second darkroom window related things
-  struct
-  {
-    GtkWidget *second_wnd;
-    GtkWidget *widget;
-    int32_t orig_width, orig_height;
-    int width, height;
-    int32_t border_size;
-    double dpi, dpi_factor, ppd, ppd_thb;
-
-    GtkWidget *button;
+  GtkWidget *second_wnd, *second_wnd_button;
 
-    dt_dev_zoom_t zoom;
-    int closeup;
-    float zoom_x, zoom_y;
-    float zoom_scale;
-
-    double button_x;
-    double button_y;
-  } second_window;
+  // several views of the same image
+  dt_dev_viewport_t full, preview2;
```

Comment:
`full` -> `preview` or `main` ? The main view is not necessary a full view.

Reply:
This was based on `DT_DEV_PIXELPIPE_FULL`. I think we are generally a bit confused about what to call the center view image. 

main/MAIN is fine by me, but maybe we should do a full scan of the codebase and make everything consistent? Including `dt_dev_process_image` and the strings in `dt_dev_pixelpipe_type_to_str` and elsewhere.  And variable names as in `const gboolean full = piece->pipe->type & DT_DEV_PIXELPIPE_FULL;`

Might be a bit of a manual job, so are we all agreed _main_ is it? @jenshannoschwalm ? @ralfbrown ? @elstoc is there a consistent term in the manual that would be convenient to be consistent with?

Reply:
`MAIN` seems to be better but i really don't care much. And having several 100 lines of code appearing in the diffs would not make this pr more readable. Also there are many places where we use naming derivates of `FULL` like in testing gbooleans. That might add to confusion ... So i would prefer to have a manual renaming after this.

Reply:
> And having several 100 lines of code appearing in the diffs would not make this pr more readable.

TRUE, but many of those lines are now already in the diff anyway (`darktable.develop->` -> `darktable.develop->full.`) so might be better to pull off the band aid now and do the testing once...

I don't really care much either of course; it is more for the _next_ person who's going to have to figure all of this out :-)

Reply:
> @elstoc is there a consistent term in the manual that would be convenient to be consistent with?

I don't think so. I'm not sure I entirely understand the context. If this is the name of the pipe that runs in the darkroom view I guess "full" isn't really accurate if we sometimes skip modules. Does that type of pipe run anywhere else? If not, is "darkroom" too obvious?

Reply:
There are three pipes/viewports in the darkroom; "preview" (top-left navigator and also used for modules that always need the full image and as placeholder while the full/center/main image recalculates), "full"/"main"/"center"/"whatever" (the one we are considering renaming) and "preview2" (the "second darkroom image window").

Reply:
Ok let's do the renaming in another PR. I do prefer `center` BTW.